examples/secret_syncer.rs

```diff
@@ -0,0 +1,120 @@
+// Demonstrates a controller some outside resource that it needs to clean up when the owner is deleted
+
+// NOTE: This is designed to demonstrate how to use finalizers, but is not in itself a good use case for them.
+// If you actually want to clean up other Kubernetes objects then you should use `ownerReferences` instead and let
+// k8s garbage collect the children.
+
+use futures::StreamExt;
+use k8s_openapi::api::core::v1::{ConfigMap, Secret};
+use kube::{
+    api::{DeleteParams, ListParams, ObjectMeta, Patch, PatchParams, Resource},
+    error::ErrorResponse,
+};
+use kube_runtime::{
+    controller::{Context, ReconcilerAction},
+    finalizer::{finalizer, Event},
+    Controller,
+};
+use snafu::{OptionExt, ResultExt, Snafu};
+use std::time::Duration;
+
+#[derive(Debug, Snafu)]
+enum Error {
+    NoName,
+    NoNamespace,
+    UpdateSecret { source: kube::Error },
+    DeleteSecret { source: kube::Error },
+}
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+fn api_in_ns_of_object<K: Resource, K2: Resource<DynamicType = ()>>(
+    object: &K,
+    kube: kube::Client,
+) -> Result<kube::Api<K2>> {
+    Ok(kube::Api::<K2>::namespaced(
+        kube,
+        object.meta().namespace.as_deref().context(NoNamespace)?,
+    ))
+}
+
+fn secret_name_for_configmap(cm_name: &str) -> String {
+    format!("cm---{}", cm_name)
+}
```

Comment:
this could be an `impl ConfigMap { fn secret_name(&self); }` to avoid the metadata name deref stuff below.
also is 3 dashes in the name there intentional?

Reply:
Yeah, wanted something that would stand out and look wrong in an actual cluster (since this is just an example, not something that anyone should actually use).